openedx/core/djangoapps/credit/email_utils.py

```diff
@@ -0,0 +1,151 @@
+"""
+This file contains utility functions which will responsible for sending emails.
+"""
+
+import os
+
+import logging
+import pynliner
+import urlparse
+import uuid
+
+from django.conf import settings
+from django.contrib.auth.models import User
+from django.contrib.staticfiles import finders
+from django.core.cache import cache
+from django.core.mail import EmailMessage
+from django.core.urlresolvers import reverse
+from django.utils.translation import ugettext as _
+
+from email.mime.image import MIMEImage
+from email.mime.multipart import MIMEMultipart
+from email.mime.text import MIMEText
+from eventtracking import tracker
+
+from edxmako.shortcuts import render_to_string
+from microsite_configuration import microsite
+from xmodule.modulestore.django import modulestore
+
+
+log = logging.getLogger(__name__)
+
+
+def send_credit_notifications(username, course_key):
+    """Sends email notification to user on different phases during credit
+    course e.g., credit eligibility, credit payment etc.
+    """
+    try:
+        user = User.objects.get(username=username)
+    except User.DoesNotExist:
+        log.error('No user with %s exist', username)
+        return
+
+    course = modulestore().get_course(course_key, depth=0)
+    course_display_name = course.display_name
+    branded_logo = dict(title='Logo', path=settings.NOTIFICATION_EMAIL_EDX_LOGO, cid=str(uuid.uuid4()))
+    tracking_context = tracker.get_tracker().resolve_context()
+    tracking_id = str(tracking_context.get('user_id'))
+    client_id = str(tracking_context.get('client_id'))
+    events = '&t=event&ec=email&ea=open'
+    tracking_pixel = 'https://www.google-analytics.com/collect?v=1&tid' + tracking_id + '&cid' + client_id + events
+    dashboard_link = _email_url_parser('dashboard')
+    credit_course_link = _email_url_parser('courses', "?type=credit")
+    context = {
+        'full_name': user.get_full_name(),
+        'platform_name': settings.PLATFORM_NAME,
+        'course_name': course_display_name,
+        'branded_logo': branded_logo['cid'],
+        'dashboard_link': dashboard_link,
+        'credit_course_link': credit_course_link,
+        'tracking_pixel': tracking_pixel,
+    }
+
+    # TODO: Update the dummy notification messages with real content
```

Comment:
Griff updated the document linked in the ticket with final copy for the eligibility email.